### PR TITLE
Document timestamp with precision

### PR DESCRIPTION
**Pull request by mosabua**

### CODE REVIEW

File: presto-docs/src/main/sphinx/language/types.rst [hunk 1]
```diff
@@ -169,22 +172,62 @@ See also :doc:`/language/timestamp`
...
 ``TIMESTAMP WITH TIME ZONE``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day with a time zone.
     Values of this type are rendered using the time zone from the value.
+    Equivalent to ``TIMESTAMP(3) WITH TIME ZONE``.
 
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321 America/Los_Angeles'``
 
+``TIMESTAMP(P) WITH TIME ZONE``
```
Comment (martint) [inline presto-docs/src/main/sphinx/language/types.rst:223]: This is not yet supported in master. It's in progress: https://github.com/prestosql/presto/pull/3947

Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:223]: oops ... let me take it back out..

Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst]: done

**Comment by mosabua** [comment]
Other questions:

All functions that support timestamp now support timestamp(p) .. so should we document that. Is it ok to just use timestamp(p) everywhere ? 

Example: at_timezone, with_timezone and others

Does current_timestamp(6) work?

**Comment by mosabua** [comment]
And for one with a return type like

from_iso8601_timestamp(string) -> timestamp with time zone

should I change them to 

from_iso8601_timestamp(string) -> timestamp(3) with time zone

**Comment by martint** [comment]
> All functions that support timestamp now support timestamp(p) .. so should we document that. Is it ok to just use timestamp(p) everywhere ?

I think that's something we mention in the release notes, but no point in documenting the fact that functions that support timestamp support timestamp :)

There's no such a thing as `timestamp` as a distinct thing from `timestamp(p)`. One is a shorthand for a special case of the other (similar to `decimal(3)` being the same as `decimal(3, 0)`)

**Comment by martint** [comment]
> Does current_timestamp(6) work?

Not yet. That's part of https://github.com/prestosql/presto/pull/3947

**Comment by martint** [comment]
> should I change them to
> from_iso8601_timestamp(string) -> timestamp(3) with time zone

I think so. It makes it clearer what precision you're getting.

**Comment by mosabua** [comment]
> > All functions that support timestamp now support timestamp(p) .. so should we document that. Is it ok to just use timestamp(p) everywhere ?
> 
> I think that's something we mention in the release notes, but no point in documenting the fact that functions that support timestamp support timestamp :)
> 
> There's no such a thing as `timestamp` as a distinct thing from `timestamp(p)`. One is a shorthand for a special case of the other (similar to `decimal(3)` being the same as `decimal(3, 0)`)

I know .. thats why I think it is misleading to leave ``timestamp`` alone, since it now actually supports not just the equivalent ``timestamp(3)`` but also ``timestamp(12)`` and so on.. hence the suggestion to be explicit and use ``timestamp(p)``

File: presto-docs/src/main/sphinx/language/types.rst [hunk 2]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    You can use ``TIMESTAMP(P)`` as an equivalent shorter name. The default
+    ``TIMESTAMP`` is an alias for ``TIMESTAMP(3)`` and means that up to
+    millisecond values are included. A precision of up to 12, so picoseconds, is
+    supported.
+
+    You can construct a timestamp value with the literal TIMESTAMP expression.
+    Alternatively, langugage constructs such as ``localtimestamp(p)``, or a
```
Comment (martint) [inline presto-docs/src/main/sphinx/language/types.rst:199]: typo: langugage

File: presto-docs/src/main/sphinx/language/types.rst [hunk 3]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    You can use ``TIMESTAMP(P)`` as an equivalent shorter name. The default
+    ``TIMESTAMP`` is an alias for ``TIMESTAMP(3)`` and means that up to
+    millisecond values are included. A precision of up to 12, so picoseconds, is
+    supported.
+
+    You can construct a timestamp value with the literal TIMESTAMP expression.
```
Comment (martint) [inline presto-docs/src/main/sphinx/language/types.rst:198]: literal TIMESTAMP -> TIMESTAMP literal

File: presto-docs/src/main/sphinx/language/types.rst [hunk 4]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+``TIMESTAMP(P) WITHOUT TIME ZONE``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    You can use ``TIMESTAMP(P)`` as an equivalent shorter name. The default
+    ``TIMESTAMP`` is an alias for ``TIMESTAMP(3)`` and means that up to
+    millisecond values are included. A precision of up to 12, so picoseconds, is
+    supported.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/language/types.rst:196]: "so picoseconds" sounds informal

File: presto-docs/src/main/sphinx/language/types.rst [hunk 5]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+
+    You can construct a timestamp value with the TIMESTAMP literal expression.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/language/types.rst:198]: wrap TIMESTAMP in ``` `` ``` ?

File: presto-docs/src/main/sphinx/language/types.rst [hunk 6]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    Alternatively, language constructs such as ``localtimestamp(p)``, or a
+    number of :doc:`date and time function and operators </functions/datetime>`
```
Comment (findepi) [inline presto-docs/src/main/sphinx/language/types.rst:200]: function -> functions

File: presto-docs/src/main/sphinx/language/types.rst [hunk 7]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    can return timestamp values.
+
+    Casting to lower precision causes rounding, and to higher precision causes
+    appended zero values.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/language/types.rst:204]: maybe explicit: "causes rounding (not truncation)"

the other part: values -> digits; but i would rather omit this.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 8]
```diff
@@ -66,8 +72,8 @@ Date and Time Functions
 .. function:: from_iso8601_timestamp(string) -> timestamp with time zone
 
     Parses the ISO 8601 formatted date ``string``, optionally with
-    time and time zone, into a ``timestamp with time zone``. The time defaults
-    to zero, and the time zone defaults to the session time zone::
+    time and time zone, into a ``timestamp with time zone``. The time
+    defaults to zero, and the time zone defaults to the session time zone::
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:76]: We had a change in here that I backed out .. but left format change. I will update the zero.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 9]
```diff
@@ -101,19 +107,22 @@ Date and Time Functions
 
     Returns a timestamp with time zone from ``timestamp`` and ``zone``.
 
-.. function:: from_unixtime(unixtime) -> timestamp
+.. function:: from_unixtime(unixtime) -> timestamp(3)
 
-    Returns the UNIX timestamp ``unixtime`` as a timestamp. ``unixtime`` is the number of seconds since ``1970-01-01 00:00:00``.
+    Returns the UNIX timestamp ``unixtime`` as a ``timestamp(3)``. ``unixtime``
+    is the number of seconds since ``1970-01-01 00:00:00``.
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:113]: Updated..

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 10]
```diff
@@ -101,19 +107,22 @@ Date and Time Functions
...
 
 .. function:: from_unixtime(unixtime, zone) -> timestamp with time zone
 
     Returns the UNIX timestamp ``unixtime`` as a timestamp with time zone
-    using ``zone`` for the time zone. ``unixtime`` is the number of seconds since ``1970-01-01 00:00:00``.
+    using ``zone`` for the time zone. ``unixtime`` is the number of seconds
+    since ``1970-01-01 00:00:00``.
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:119]: Add info

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 11]
```diff
@@ -101,19 +107,22 @@ Date and Time Functions
...
 
 .. function:: from_unixtime(unixtime, hours, minutes) -> timestamp with time zone
 
     Returns the UNIX timestamp ``unixtime`` as a timestamp with time zone
-    using ``hours`` and ``minutes`` for the time zone offset. ``unixtime`` is the number of seconds since ``1970-01-01 00:00:00``.
+    using ``hours`` and ``minutes`` for the time zone offset. ``unixtime`` is
+    the number of seconds since ``1970-01-01 00:00:00``.
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:125]: Done

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 12]
```diff
@@ -123,6 +132,14 @@ Date and Time Functions
 
     Returns the current timestamp as of the start of the query.
 
+.. data:: localtimestamp(p)
+
+    Returns the current :ref:`timestamp <timestamp-data-type>` as of the start
+    of the query, with ``p`` digits of subsecond precision::
+
+        SELECT localtimestamp(6);
+        -- 2020-06-10 15:55:23.383628
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:141]: We decided against that for now..

File: presto-docs/src/main/sphinx/language/types.rst (hunk 4 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:196]: yes.. and that is fine.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 13]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321'``
 
+``TIMESTAMP(P) WITHOUT TIME ZONE``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    You can use ``TIMESTAMP(P)`` as an equivalent shorter name. The default
+    ``TIMESTAMP`` is an alias for ``TIMESTAMP(3)`` and means that up to
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:194]: Its consistent in this doc .. if we change this we should do it as separate bulk change. So not in this PR.

File: presto-docs/src/main/sphinx/language/types.rst (hunk 5 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:198]: Done

File: presto-docs/src/main/sphinx/language/types.rst (hunk 6 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:200]: done

File: presto-docs/src/main/sphinx/language/types.rst (hunk 7 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:204]: Lets leave it in to be explicit.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 14]
```diff
@@ -2,6 +2,12 @@
 Date and Time Functions and Operators
 =====================================
 
+Use the date and time functions and operators, detailed in the following, to
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:5]: We don't normally give directives in the documentation. Instead, phrase this as

```
These functions and operators operate on :ref:`date and time data types <date-time-data-types>`.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 15]
```diff
@@ -66,8 +72,9 @@ Date and Time Functions
 .. function:: from_iso8601_timestamp(string) -> timestamp with time zone
 
     Parses the ISO 8601 formatted date ``string``, optionally with
-    time and time zone, into a ``timestamp with time zone``. The time defaults
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:69]: This seems to be just changing the wrapping. Please remove

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 16]
```diff
@@ -101,19 +108,22 @@ Date and Time Functions
 
     Returns a timestamp with time zone from ``timestamp`` and ``zone``.
 
-.. function:: from_unixtime(unixtime) -> timestamp
+.. function:: from_unixtime(unixtime) -> timestamp(3)
 
-    Returns the UNIX timestamp ``unixtime`` as a timestamp. ``unixtime`` is the number of seconds since ``1970-01-01 00:00:00``.
+    Returns the UNIX timestamp ``unixtime`` as a ``timestamp(3)``. ``unixtime``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:113]: Leave this part alone. It's fine to say "as a timestamp" since the function definition gives the exact type.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 17]
```diff
@@ -101,19 +108,22 @@ Date and Time Functions
...
+    is the number of seconds since ``1970-01-01 00:00:00 UTC``.
 
 .. function:: from_unixtime(unixtime, zone) -> timestamp with time zone
 
-    Returns the UNIX timestamp ``unixtime`` as a timestamp with time zone
-    using ``zone`` for the time zone. ``unixtime`` is the number of seconds since ``1970-01-01 00:00:00``.
+    Returns the UNIX timestamp ``unixtime``, a double as a timestamp with
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:118]: This seems unrelated to timestamp with precision. Please remove.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 18]
```diff
@@ -101,19 +108,22 @@ Date and Time Functions
...
+    time zone using ``zone`` for the time zone. ``unixtime`` is the number of
+    seconds since ``1970-01-01 00:00:00 UTC`` in ``double`` data type.
 
 .. function:: from_unixtime(unixtime, hours, minutes) -> timestamp with time zone
 
     Returns the UNIX timestamp ``unixtime`` as a timestamp with time zone
-    using ``hours`` and ``minutes`` for the time zone offset. ``unixtime`` is the number of seconds since ``1970-01-01 00:00:00``.
+    using ``hours`` and ``minutes`` for the time zone offset. ``unixtime`` is
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:125]: Same

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 19]
```diff
@@ -220,6 +238,12 @@ Unit              Description
         SELECT date_diff('day', DATE '2020-03-01', DATE '2020-03-02');
         -- 1
 
+        SELECT date_diff('second', TIMESTAMP '2020-06-01 12:30:45.000000000',TIMESTAMP '2020-06-02 12:30:45.123456789');
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:241]: Add space after comma

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 20]
```diff
@@ -307,9 +331,9 @@ Specifier Description
 
     Formats ``timestamp`` as a string using ``format``.
 
-.. function:: date_parse(string, format) -> timestamp
+.. function:: date_parse(string, format) -> timestamp(3)
 
-    Parses ``string`` into a timestamp using ``format``.
+    Parses ``string`` into a timestamp(3) using ``format``.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:336]: Remove the precision here, since this is a word, not a type.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 21]
```diff
@@ -141,6 +141,9 @@ String
     JSON value type, which can be a JSON object, a JSON array, a JSON number, a JSON string,
     ``true``, ``false`` or ``null``.
 
+
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:144]: Remove extra blank line

File: presto-docs/src/main/sphinx/language/types.rst [hunk 22]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
 
     Example: ``TIME '01:02:03.456 America/Los_Angeles'``
 
+.. _timestamp-data-type:
+
 ``TIMESTAMP``
 ^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day without a time zone.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:180]: Remove the duplicate description. This section should only mention that this is an alias.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 23]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
     Values of this type are parsed and rendered in the session time zone.
+    Equivalent to ``TIMESTAMP WITHOUT TIME ZONE``, ``TIMESTAMP(3)``, and
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:182]: We don't need to mention all the various redundant syntaxes.
```
``TIMESTAMP`` is an alias for ``TIMESTAMP(3)``.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 24]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    ``TIMESTAMP(3) WITHOUT TIME ZONE``.
 
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321'``
 
+``TIMESTAMP(P) WITHOUT TIME ZONE``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:187]: Change this to just ``TIMESTAMP(P)``. We can mention in the description that this type can also be written as ``TIMESTAMP WITHOUT TIME ZONE``. However, the canonical name for the type is just ``TIMESTAMP``, when rendered by the SQL engine (e.g., in ``DESCRIBE``) output, so we should call it that here.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 25]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
 
     Example: ``TIME '01:02:03.456 America/Los_Angeles'``
 
+.. _timestamp-data-type:
+
 ``TIMESTAMP``
 ^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day without a time zone.
     Values of this type are parsed and rendered in the session time zone.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:181]: We are missing this second sentence in the description below.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 26]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
 
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321'``
 
+``TIMESTAMP(P) WITHOUT TIME ZONE``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    You can use ``TIMESTAMP(P)`` as an equivalent shorter name. The default
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:193]: This should be the opposite. ``WITHOUT TIME ZONE`` is the equivalent longer name.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 27]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    ``TIMESTAMP`` is an alias for ``TIMESTAMP(3)`` and means that up to
+    millisecond values are included. A precision of up to 12, so picoseconds, is
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:195]: > A precision of up to 12 (picoseconds) is supported.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 28]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    supported.
+
+    You can construct a timestamp value with the ``TIMESTAMP`` literal
+    expression. Alternatively, language constructs such as
+    ``localtimestamp(p)``, or a number of :doc:`date and time functions and
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:200]: It's best not to wrap in the middle of a `:doc:` reference. Better to put it on it's own line so the entire thing can be read and verified in its entirety.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 29]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    operators </functions/datetime>` can return timestamp values.
+
+    Casting to lower precision causes rounding, and not truncating. Casting to
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:203]: "truncating" sounds weird. Let's say

> causes the value to be rounded (not truncated).

File: presto-docs/src/main/sphinx/language/types.rst [hunk 30]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+    higher precision causes appended zero digits.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:204]: > Casting to higher precision appends zeros for the additional digits.

File: presto-docs/src/main/sphinx/language/types.rst [hunk 31]
```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
...
+
+        SELECT cast(TIMESTAMP '2020-06-10 15:55:23.383345' as TIMESTAMP(12));
+        -- 2020-06-10 15:55:23.383345000000
+
 ``TIMESTAMP WITH TIME ZONE``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day with a time zone.
     Values of this type are rendered using the time zone from the value.
+    Equivalent to ``TIMESTAMP(3) WITH TIME ZONE``.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/language/types.rst:225]: Remove, since this is not implemented yet.

File: presto-docs/src/main/sphinx/language/types.rst (hunk 26 above)
Comment (martint) [reply presto-docs/src/main/sphinx/language/types.rst:193]: Technically, no. The canonical name for the type in the SQL spec is “timestamp without time zone”, even though we use the short name everywhere in Presto

File: presto-docs/src/main/sphinx/language/types.rst (hunk 24 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/language/types.rst:187]: Also, from purely human (non-scientific) perspective, `TIMESTAMP WITHOUT TIME ZONE` looks very similar to `TIMESTAMP WITH TIME ZONE` (eg has levenshtein distance of 3 with length of 27, being 88% similar).  `TIMESTAMP` is much easier to read.

File: presto-docs/src/main/sphinx/language/types.rst (hunk 26 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/language/types.rst:193]: You can assume my comment means "in Presto".

The the documentation should reflect the actual usage. e.g., same with `double precision`. We document what exists, not what would exist if we exactly followed the specification.

While I can see an argument for changing everything to use the longer name, that also seems like a tradeoff between readability/usability and being pedantic. (PostgreSQL seems to have chosen the latter in this case)

File: presto-docs/src/main/sphinx/language/types.rst (hunk 28 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:200]: I just use the wrapping of VisualCode to the 80 char and it works just fine. Do you really want me to manually wrap such occurrences..

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 16 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:113]: @martint wanted to make sure we are explicit .. so I added it as he requested. Personally I think this helps. Can you guys decide together please..

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 14 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:5]: I would be more approachable to address the user .. do you still want me to rejig?

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 15 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:69]: It adds the milliseconds for correctness as requested by @martint

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 16 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:113]: @martint specifically requested this change on all outputs ..

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 17 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:118]: This add UTC as requested by @martint

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 18 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:125]: this add the data type as requested by @findepi

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 19 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:241]: done

File: presto-docs/src/main/sphinx/language/types.rst (hunk 21 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/language/types.rst:144]: done

File: presto-docs/src/main/sphinx/language/types.rst (hunk 24 above)
Comment (martint) [reply presto-docs/src/main/sphinx/language/types.rst:187]: Yes, that bit me a few times when writing the code to support these types. It's easy to get lost in a sea of names when you have things like`TimestampWithoutTimeZoneToTimestampWithTimeZoneCast`. (I didn't go that route, fwiw)

File: presto-docs/src/main/sphinx/language/types.rst (hunk 26 above)
Comment (martint) [reply presto-docs/src/main/sphinx/language/types.rst:192]: I'm not suggesting we change the names to the longer form. I was just commenting on the fact that the canonical name is actually the long form and the other version is just an alias, for the benefit of others that might see this discussion in the future.

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 18 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/functions/datetime.rst:125]: I am out of context. what did i request that @electrum does not enjoy?

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 14 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/datetime.rst:5]: Yes. We want to keep the style consistent.

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 15 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/datetime.rst:69]: This formatting change makes it hard to see what actually changed. Please leave it alone except for adding the fractional seconds.

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 17 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/datetime.rst:116]: Please only add "UTC" and don't make any formatting changes. Adding data types is unrelated to timestamp precision and should be done / discussed separately.

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 18 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/datetime.rst:123]: Please revert this change as it's unrelated to timestamp precision.

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 32]
```diff
@@ -123,6 +133,14 @@ Date and Time Functions
 
     Returns the current timestamp as of the start of the query.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst:132]: We could add
```
, with ``3`` digits of subsecond precision.

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 20 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/datetime.rst:334]: See comment. Please remove precision.

File: presto-docs/src/main/sphinx/language/types.rst (hunk 22 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/language/types.rst:179]: See comment

File: presto-docs/src/main/sphinx/language/types.rst (hunk 25 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/language/types.rst:180]: See comment

File: presto-docs/src/main/sphinx/language/types.rst (hunk 23 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/language/types.rst:181]: See comment

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 15 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:69]: done

File: presto-docs/src/main/sphinx/language/types.rst (hunk 28 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/language/types.rst:199]: Yes, we format for readability, not arbitrary wrapping at 80 characters

File: presto-docs/src/main/sphinx/functions/datetime.rst (hunk 32 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/functions/datetime.rst:132]: added
